libs/Facilitator/Facilitator.js

```diff
@@ -0,0 +1,387 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const BN = require('bn.js');
+const web3 = require('web3');
+const Contracts = require('../../libs/Contracts');
+const Utils = require('../utils/Utils.js');
+
+/**
+ * Class to facilitate stake and mint.
+ *
+ * @class
+ */
+class Facilitator {
+  /**
+   * Constructor for facilitator.
+   * @constructor
+   *
+   * @param {Object} originWeb3 Origin chain web3 object.
+   * @param {Object} auxiliaryWeb3 Auxiliary chain web3 object.
+   * @param {string} Gateway contract address.
+   * @param {Object} CoGateway contract address.
+   */
+  constructor(originWeb3, auxiliaryWeb3, gatewayAddress, coGatewayAddress) {
+    if (originWeb3 === undefined) {
+      throw new Error('Invalid origin web3 object.');
+    }
+
+    if (auxiliaryWeb3 === undefined) {
+      throw new Error('Invalid auxiliary web3 object.');
+    }
+
+    if (!web3.utils.isAddress(gatewayAddress)) {
+      throw new Error('Invalid Gateway address.');
+    }
+
+    if (!web3.utils.isAddress(coGatewayAddress)) {
+      throw new Error('Invalid Cogateway address.');
+    }
+
+    this.originWeb3 = originWeb3;
+    this.auxiliaryWeb3 = auxiliaryWeb3;
+    this.contracts = new Contracts(originWeb3, auxiliaryWeb3);
+    this.gatewayAddress = gatewayAddress;
+    this.coGatewayAddress = coGatewayAddress;
+  }
+
+  /**
+   * @function stake
+   *
+   * Performs the stake process.
+   *
+   * @param {string} staker Staker address.
+   * @param {string} amount Stake amount.
+   * @param {string} beneficiary Beneficiary address for minting tokens.
+   * @param {string} gasPrice Gas price for reward calculation.
+   * @param {string} gasLimit Mas gas for reward calculation.
+   * @param {string} unlockSecret Unlock secret that will be used for progress stake.
+   * @param {string} gas The gas provided for the transaction execution
+   *
+   * @returns {Promise} Promise object.
+   */
+  async stake(
+    staker,
+    amount,
+    beneficiary,
+    gasPrice = '0',
+    gasLimit = '0',
+    unlockSecret,
+    facilitator,
+    gas = '7000000'
```

Comment:
Where does this number come from? (`7000000`)

Reply:
Removed this.